Remove most @deprecated

dwijnand opened the pull request:
Some have become private[sbt] instead.

Additionally:
* space out terms
* remove commented-out blocks
* hand format long lines
* some general cleanup

Closes #3078 (not all are removed, but enough to close the issue)

eed3si9n reviewed (changes requested):
A bunch of failures on Travis. Some of them I'm sure are legit.

```
[error] 	actions / cross-multiproject
[error] 	actions / cross
[error] 	actions / depends-on
[error] 	actions / state
[error] 	actions / update-state-fail
[error] 	apiinfo / unstable-existential-names
[error] 	compiler-project / inc-pickled-existential
[error] 	compiler-project / inc-pickled-refinement
[error] 	compiler-project / src-dep-plugin
[error] 	dependency-management / ivy-settings-a
[error] 	dependency-management / ivy-settings-multi-b
[error] 	dependency-management / override
[error] 	dependency-management / override2
[error] 	dependency-management / pom-advanced
[error] 	dependency-management / pom-classpaths
[error] 	dependency-management / provided
[error] 	project / load-hooks
[error] 	project / scripted-plugin
[error] 	project / src-plugins
[error] 	source-dependencies / abstract-type-override
[error] 	tests / empty
[error] 	tests / fork
[error] 	tests / serial
```

dwijnand committed:
Just whitespace & comment changes

dwijnand committed:
Code formatting only changes

dwijnand committed:
Simplify sbt.Compiler

dwijnand committed:
Drop deprecated SubProcess apply overload

dwijnand committed:
Drop most remaining deprecations

dwijnand committed:
Drop deprecated InputTask apply method

dwijnand committed:
Drop ProjectReference implicit lifts

dwijnand committed:
Drop File/Seq[File] setting enrichments

dwijnand committed:
Drop deprecated seq method

dwijnand committed:
Drop deprecated inputTask method

Temporarily (FLW) duplicate it in fullRunInputTask.

dwijnand committed:
Define DefaultEvaluateTaskConfig

dwijnand commented:
* reverted a change in BuildUtil.aggregationRelation (left a comment) which was breaking actions/cross-multiproject and actions/update-state-fail
* dropped the attempt to remove the deprecated methods in scripted because I can't figure out how to fix project/scripted-plugin
* dropped the attempt to remove the deprecated DefinableTask#task because I can't figure out how to fix actions/depends-on
* fixed all the other broken scripted tests due to the deprecation removal

If Travis CI passes I'll squash this down.

dwijnand committed:
Drop deprecated default resolvers

eed3si9n commented on main-settings/src/main/scala/sbt/Reference.scala:
```diff
@@ -94,13 +94,4 @@ object Reference {
   }
-
-  @deprecated("Explicitly wrap the URI in a call to RootProject.", "0.13.0")
-  implicit def uriToRef(u: URI): ProjectReference = RootProject(u)
```
Should some of these survive as Restligeist macro?

dwijnand replied:
I think 4 years is sufficient time to allow users to migrate before we drop the maintenance cost.. https://github.com/sbt/sbt/commit/0507a2a1a9be5006b6ffb4dee3318796b70fe290

eed3si9n commented:
I know we're technically just removing deprecations, but maybe some of the popularly used methods should be mentioned in the migration guide and/or Restligeist macro.

dwijnand committed:
Detail removal of commonly used deprecations in MIGRATION.md
